eZ/Publish/API/Repository/Tests/SearchService/SortClause/ScoreTest.php

```diff
@@ -0,0 +1,168 @@
+    {
+        $this->createContentForScoreSortTesting($inputValues);
+
+        $searchService = $this->getRepository()->getSearchService();
+        if ($query instanceof LocationQuery) {
+            $actualResults = $searchService->findLocations($query);
+        } else {
+            $actualResults = $searchService->findContentInfo($query);
+        }
+
+        $this->assertSearchResultOrderByRemoteId($expectedOrderedIds, $actualResults);
+    }
+
+    public function dataProviderForTestSortingByScore(): iterable
+    {
+        // The following input values for test content guarantee predictable scoring
+        $inputValues = ['foo foo', 'foo', 'foo foo foo'];
+
+        yield 'content asc' => [
+            $inputValues,
+            new Query([
+                'query' => new Criterion\FullText('foo'),
+                'sortClauses' => [
+                    new SortClause\Score(Query::SORT_ASC),
+                    new SortClause\ContentId(),
+                ],
+            ]),
+            ['foo', 'foo foo', 'foo foo foo'],
+        ];
+
+        yield 'content desc' => [
+            $inputValues,
+            new Query([
+                'query' => new Criterion\FullText('foo'),
+                'sortClauses' => [
+                    new SortClause\Score(Query::SORT_DESC),
+                    new SortClause\ContentId(),
+                ],
+            ]),
+            ['foo foo foo', 'foo foo', 'foo'],
+        ];
+
+        yield 'location asc' => [
+            $inputValues,
+            new LocationQuery([
+                'query' => new Criterion\FullText('foo'),
+                'sortClauses' => [
+                    new SortClause\Score(Query::SORT_ASC),
+                    new SortClause\ContentId(),
+                ],
+            ]),
+            ['foo', 'foo foo', 'foo foo foo'],
+        ];
+
+        yield 'location desc' => [
+            $inputValues,
+            new LocationQuery([
+                'query' => new Criterion\FullText('foo'),
+                'sortClauses' => [
+                    new SortClause\Score(Query::SORT_DESC),
+                    new SortClause\ContentId(),
+                ],
+            ]),
+            ['foo foo foo', 'foo foo', 'foo'],
+        ];
+    }
+
+    private function assertSearchResultOrderByRemoteId(
+        array $expectedOrderedIds,
+        SearchResult $actualSearchResults
+    ): void {
+        self::assertEquals(
+            count($expectedOrderedIds),
+            $actualSearchResults->totalCount
+        );
+
+        $actualIds = array_map(
+            static function (SearchHit $searchHit): string {
+                return $searchHit->valueObject->remoteId;
+            },
+            $actualSearchResults->searchHits
+        );
+
+        self::assertEquals($expectedOrderedIds, $actualIds);
+    }
+
+    private function createContentForScoreSortTesting(iterable $values): void
```

Comment:
Here we can report common Repository exception`\eZ\Publish\API\Repository\Exceptions\Exception`, since there are multiple unhandled exceptions in this method.